tokio-rustls/tests/badssl.rs

```diff
@@ -95,3 +100,79 @@ async fn test_modern() -> io::Result<()> {
 
     Ok(())
 }
+
+#[tokio::test]
+async fn test_client() -> io::Result<()> {
+    let cert = certs(&mut io::BufReader::new(CLIENT_CERT.as_bytes()))
+        .unwrap()
+        .drain(..)
+        .map(rustls::Certificate)
+        .collect();
+
+    let mut keys = rsa_private_keys(&mut io::BufReader::new(io::Cursor::new(CLIENT_RSA))).unwrap();
+    let mut keys = keys.drain(..).map(PrivateKey);
+
+    let mut root_store = rustls::RootCertStore::empty();
+    root_store.add_server_trust_anchors(webpki_roots::TLS_SERVER_ROOTS.0.iter().map(|ta| {
+        OwnedTrustAnchor::from_subject_spki_name_constraints(
+            ta.subject,
+            ta.spki,
+            ta.name_constraints,
+        )
+    }));
+
+    let config = rustls::ClientConfig::builder()
+        .with_safe_defaults()
+        .with_root_certificates(root_store)
+        .with_single_cert(cert, keys.next().unwrap())
+        .unwrap();
+    let config = Arc::new(config);
+    let domain = "client.badssl.com";
+
+    let (_, output) = get(config.clone(), domain, 443).await?;
```

Comment:
Nits: there's no need to clone here. Maybe inline the `Arc::new()` and domain here?